src/Tizen.Applications.WidgetControl/Tizen.Applications/WidgetControl.cs

```diff
@@ -314,6 +314,74 @@ public static string[] GetWidgetIds(string pkgId)
             return list.ToArray();
         }
 
+        /// <summary>
+        /// Gets main appid of the widget.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <privilege>http://tizen.org/privilege/widget.viewer</privilege>
+        /// <exception cref="ArgumentException">Thrown when failed because of an invalid argument.</exception>
+        /// <exception cref="InvalidOperationException">Thrown in case of failed conditions.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the API is not supported in this device.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when an application does not have the privilege to access this method.</exception>
+        public static string GetWidgetMainAppId(string widgetId)
+        {
+            string appid = Interop.WidgetService.GetWidgetMainAppId(widgetId);
+            Interop.WidgetService.ErrorCode err = (Interop.WidgetService.ErrorCode)Internals.Errors.ErrorFacts.GetLastResult();
+            switch (err)
+            {
+                case Interop.WidgetService.ErrorCode.InvalidParameter:
+                    throw new ArgumentException("Invalid parameter");
+
+                case Interop.WidgetService.ErrorCode.NotExist:
+                    throw new InvalidOperationException("widget id is not exist");
+
+                case Interop.WidgetService.ErrorCode.IoError:
+                case Interop.WidgetService.ErrorCode.Fault:
+                    throw new InvalidOperationException("IO error");
+
+                case Interop.WidgetService.ErrorCode.PermissionDenied:
+                    throw new UnauthorizedAccessException();
+
+                case Interop.WidgetService.ErrorCode.NotSupported:
+                    throw new NotSupportedException();
+            }
+            return appid;
+        }
+
+        /// <summary>
+        /// Gets package ID of the widget.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <privilege>http://tizen.org/privilege/widget.viewer</privilege>
+        /// <exception cref="ArgumentException">Thrown when failed because of an invalid argument.</exception>
+        /// <exception cref="InvalidOperationException">Thrown in case of failed conditions.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the API is not supported in this device.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when an application does not have the privilege to access this method.</exception>
+        public static string GetWidgetPackageId(string widgetId)
```

Comment:
How about add PackageId property on WidgetControl instead of this method?
``` c#
class WidgetControl {
     public string PackageId 
    { 
         get 
         {
              return Interop.WidgetService.GetWidgetPackageId(Id);
         }
     }
}
```

Reply:
First, What is widget object? which class was represented it?
Why don't you add a property `PackageId` on `widget object`?

Reply:
My explanation was poor, created widget control object is designated for only one widget id but some service just want to get multiple widget's package id and i think it is waste of resources if it create every widget id for just get package id.

Reply:
Hmm, this is for supporting "widget_service_get_package_id()" as C# API. Other attributes of widget which can access with widget_id are supported by property of WidgetControl class. I think this needs to be property also for integrity.